code/datums/components/area_sound_manager.dm

```diff
@@ -0,0 +1,77 @@
+///Allows you to set a theme for a set of areas without tying them to looping sounds explicitly
+/datum/component/area_sound_manager
+	///area -> looping sound type
+	var/list/area_to_looping_type = list()
+	///Current sound loop
+	var/datum/looping_sound/our_loop
+	///A list of "acceptable" z levels to be on. If you leave this, we're gonna delete ourselves
+	var/list/accepted_zs
+	///The timer id of our current start delay, if it exists
+	var/timerid
+
+/datum/component/area_sound_manager/Initialize(area_loop_pairs, change_on, remove_on, acceptable_zs)
+	if(!ismovable(parent))
+		return
+	area_to_looping_type = area_loop_pairs
+	accepted_zs = acceptable_zs
+	change_the_track()
+
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/react_to_move)
```

Comment:
![image](https://user-images.githubusercontent.com/58055496/119428385-9f2a6d80-bcc1-11eb-8e88-575167bf7204.png)
Discussed on discord, long and short of it is Entered isn't currently called on areas by abstract movement, the sort ghosts use. That might change, but it's not a huge issue here because of how my code is constructed